aodn_cloud_optimised/bin/generic_cloud_optimised_creation.py

```diff
@@ -49,131 +27,236 @@
 from aodn_cloud_optimised.lib.s3Tools import s3_ls
 
 
+# ---------- Pydantic schema for validating the config ----------
+class ClusterConfig(BaseModel):
+    mode: Optional[str] = None
+    restart_every_path: bool = False
+
+
+class PathConfig(BaseModel):
+    s3_uri: str
+    filter: Optional[List[str]] = Field(default_factory=list)
```

Comment:
```suggestion
    filters: Optional[List[str]] = Field(default=[], description="A list of regex filters to apply to the X")
```

I think this is a simpler, should be filter*s* as it is a list and could have many.

Not sure actually what is being filtered here